src/main/java/com/odysseusinc/arachne/executionengine/service/versiondetector/DefaultVersionDetectionService.java

```diff
@@ -0,0 +1,154 @@
+/*
+ *
+ * Copyright 2019 Odysseus Data Services, inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Company: Odysseus Data Services, Inc.
+ * Product Owner/Architecture: Gregory Klebanov
+ * Authors: Pavel Grafkin, Vitaly Koulakov, Anastasiia Klochkova, Yaroslav Molodkov, Alexander Cumarav
+ * Created: October 17, 2019
+ *
+ */
+
+package com.odysseusinc.arachne.executionengine.service.versiondetector;
+
+import com.odysseusinc.arachne.commons.types.CommonCDMVersionDTO;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.DataSourceUnsecuredDTO;
+import com.odysseusinc.arachne.executionengine.util.SQLUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Service
+public class DefaultVersionDetectionService extends BaseVersionDetectionService implements VersionDetectionService {
+
```

Comment:
I don't really like the format of the cdm_detection_notes.txt file is it possible use the line-by-line format like we do for logging:
```[V4_0] Database table source_to_concept_map  missed fields: source_code, source_vocabulary_id, source_code_description, target_concept_id, target_vocabulary_id, mapping_type, primary_map, valid_start_date, valid_end_date, invalid_reason
[V4_0] Database table vocabulary  missed fields: vocabulary_id, vocabulary_name
[V4_0] Database table drug_strength  missed fields: drug_concept_id, ingredient_concept_id, amount_value, amount_unit, concentration_value, concentration_enum_unit, concentration_denom_unit, valid_start_date, valid_end_date, invalid_reason
[V4_0] Database table concept  missed fields: concept_level, concept_class
[V4_0] Database table relationship  missed fields: relationship_id, relationship_name, is_hierarchical, defines_ancestry, reverse_relationship
[V4_0] Database table concept_synonym  missed fields: concept_synonym_id
[V6_0] Database table source_to_concept_map  missed fields: source_code, source_concept_id, source_vocabulary_id, source_code_description, target_concept_id, target_vocabulary_id, valid_start_date, valid_end_date, invalid_reason
[V6_0] Database table note  missed fields: note_id, person_id, note_event_id, note_event_field_concept_id, note_date, note_datetime, note_type_concept_id, note_class_concept_id, note_title, note_text, encoding_concept_id, language_concept_id, provider_id, visit_occurrence_id, visit_detail_id, note_source_value
[V6_0] Database table metadata  missed fields: metadata_concept_id, metadata_type_concept_id, name, value_as_string, value_as_concept_id, metadata_date, metadata_datetime
[V6_0] Database table survey_conduct  missed fields: survey_conduct_id, person_id, survey_concept_id, survey_start_date, survey_start_datetime, survey_end_date, survey_end_datetime, provider_id, assisted_concept_id, respondent_type_concept_id, timing_concept_id, collection_method_concept_id, assisted_source_value, respondent_type_source_value, timing_source_value, collection_method_source_value, survey_source_value, survey_source_concept_id, survey_source_identifier, validated_survey_concept_id, validated_survey_source_value, survey_version_number, visit_occurrence_id, visit_detail_id, response_visit_occurrence_id
[V6_0] Database table drug_strength  missed fields: drug_concept_id, ingredient_concept_id, amount_value, amount_unit_concept_id, numerator_value, numerator_unit_concept_id, denominator_value, denominator_unit_concept_id, box_size, valid_start_date, valid_end_date, invalid_reason
[V6_0] Database table drug_era  missed fields: drug_era_id, person_id, drug_concept_id, drug_era_start_datetime, drug_era_end_datetime, drug_exposure_count, gap_days
[V6_0] Database table measurement  missed fields: measurement_id, person_id, measurement_concept_id, measurement_date, measurement_datetime, measurement_time, measurement_type_concept_id, operator_concept_id, value_as_number, value_as_concept_id, unit_concept_id, range_low, range_high, provider_id, visit_occurrence_id, visit_detail_id, measurement_source_value, measurement_source_concept_id, unit_source_value, value_source_value
[V6_0] Database table vocabulary  missed fields: vocabulary_id, vocabulary_name, vocabulary_reference, vocabulary_version, vocabulary_concept_id
[V6_0] Database table provider  missed fields: provider_id, provider_name, npi, dea, specialty_concept_id, care_site_id, year_of_birth, gender_concept_id, provider_source_value, specialty_source_value, specialty_source_concept_id, gender_source_value, gender_source_concept_id
[V6_0] Database table specimen  missed fields: specimen_id, person_id, specimen_concept_id, specimen_type_concept_id, specimen_date, specimen_datetime, quantity, unit_concept_id, anatomic_site_concept_id, disease_status_concept_id, specimen_source_id, specimen_source_value, unit_source_value, anatomic_site_source_value, disease_status_source_value
[V6_0] Database table visit_occurrence  missed fields: visit_occurrence_id, person_id, visit_concept_id, visit_start_date, visit_start_datetime, visit_end_date, visit_end_datetime, visit_type_concept_id, provider_id, care_site_id, visit_source_value, visit_source_concept_id, admitted_from_concept_id, admitted_from_source_value, discharge_to_source_value, discharge_to_concept_id, preceding_visit_occurrence_id
[V6_0] Database table procedure_occurrence  missed fields: procedure_occurrence_id, person_id, procedure_concept_id, procedure_date, procedure_datetime, procedure_type_concept_id, modifier_concept_id, quantity, provider_id, visit_occurrence_id, visit_detail_id, procedure_source_value, procedure_source_concept_id, modifier_source_value
[V6_0] Database table relationship  missed fields: relationship_id, relationship_name, is_hierarchical, defines_ancestry, reverse_relationship_id, relationship_concept_id
[V6_0] Database table cost  missed fields: cost_event_field_concept_id, payer_plan_period_id
[V6_0] Database table condition_era  missed fields: condition_era_start_datetime, condition_era_end_datetime
[V6_0] Database table visit_detail  missed fields: visit_detail_id, person_id, visit_detail_concept_id, visit_detail_start_date, visit_detail_start_datetime, visit_detail_end_date, visit_detail_end_datetime, visit_detail_type_concept_id, provider_id, care_site_id, discharge_to_concept_id, admitted_from_concept_id, admitted_from_source_value, visit_detail_source_value, visit_detail_source_concept_id, discharge_to_source_value, preceding_visit_detail_id, visit_detail_parent_id, visit_occurrence_id
[V6_0] Database table note_nlp  missed fields: note_nlp_id, note_id, section_concept_id, snippet, offset, lexical_variant, note_nlp_concept_id, nlp_system, nlp_date, nlp_datetime, term_exists, term_temporal, term_modifiers, note_nlp_source_concept_id
[V6_0] Database table observation  missed fields: observation_id, person_id, observation_concept_id, observation_date, observation_datetime, observation_type_concept_id, value_as_number, value_as_string, value_as_concept_id, qualifier_concept_id, unit_concept_id, provider_id, visit_occurrence_id, visit_detail_id, observation_source_value, observation_source_concept_id, unit_source_value, qualifier_source_value, observation_event_id, obs_event_field_concept_id, value_as_datetime
[V6_0] Database table observation_period  missed fields: observation_period_id, person_id, observation_period_start_date, observation_period_end_date, period_type_concept_id
[V6_0] Database table dose_era  missed fields: dose_era_start_datetime, dose_era_end_datetime
[V6_0] Database table payer_plan_period  missed fields: payer_plan_period_id, person_id, contract_person_id, payer_plan_period_start_date, payer_plan_period_end_date, payer_concept_id, plan_concept_id, contract_concept_id, sponsor_concept_id, stop_reason_concept_id, payer_source_value, payer_source_concept_id, plan_source_value, plan_source_concept_id, contract_source_value, contract_source_concept_id, sponsor_source_value, sponsor_source_concept_id, family_source_value, stop_reason_source_value, stop_reason_source_concept_id
[V6_0] Database table drug_exposure  missed fields: drug_exposure_id, person_id, drug_concept_id, drug_exposure_start_date, drug_exposure_start_datetime, drug_exposure_end_date, drug_exposure_end_datetime, verbatim_end_date, drug_type_concept_id, stop_reason, refills, quantity, days_supply, sig, route_concept_id, lot_number, provider_id, visit_occurrence_id, visit_detail_id, drug_source_value, drug_source_concept_id, route_source_value, dose_unit_source_value
[V6_0] Database table person  missed fields: person_id, gender_concept_id, year_of_birth, month_of_birth, day_of_birth, birth_datetime, death_datetime, race_concept_id, ethnicity_concept_id, location_id, provider_id, care_site_id, person_source_value, gender_source_value, gender_source_concept_id, race_source_value, race_source_concept_id, ethnicity_source_value, ethnicity_source_concept_id
[V6_0] Database table location_history  missed fields: location_history_id, location_id, relationship_type_concept_id, domain_id, entity_id, start_date, end_date
[V6_0] Database table location  missed fields: location_id, address_1, address_2, city, state, zip, county, country, location_source_value, latitude, longitude
[V6_0] Database table fact_relationship  missed fields: domain_concept_id_1, fact_id_1, domain_concept_id_2, fact_id_2, relationship_concept_id
```

Reply:
Done: 
![image](https://user-images.githubusercontent.com/7705583/82320936-608a1500-99d4-11ea-95b3-7965e1952d7d.png)